test/distributed/org/apache/cassandra/distributed/test/repair/AutoRepairSchedulerTest.java

```diff
@@ -112,17 +134,41 @@ public void testScheduler() throws ParseException
 
         // validate that the repair ran on all nodes
         cluster.forEach(i -> i.runOnInstance(() -> {
-            Util.spinAssert("AutoRepair has not yet completed one FULL repair cycle",
+            // Reduce sleeping if repair finishes quickly to speed up test but make it non-zero to provoke some
+            // contention.
+            AutoRepair.SLEEP_IF_REPAIR_FINISHES_QUICKLY = new DurationSpec.IntSecondsBound("1s");
+
+            AutoRepairMetrics incrementalMetrics = AutoRepairMetricsManager.getMetrics(AutoRepairConfig.RepairType.INCREMENTAL);
+            Util.spinAssert("AutoRepair has not yet completed one INCREMENTAL repair cycle",
                             greaterThan(0L),
-                            AutoRepair.instance.repairStates.get(AutoRepairConfig.RepairType.FULL)::getLastRepairTime,
+                            () -> incrementalMetrics.nodeRepairTimeInSec.getValue().longValue(),
                             5,
                             TimeUnit.MINUTES);
-            Util.spinAssert("AutoRepair has not yet completed one INCREMENTAL repair cycle",
+
+            // Expect some contention on incremental repair.
+            Util.spinAssert("AutoRepair has not observed any replica contention in INCREMENTAL repair",
+                            greaterThan(0L),
+                            incrementalMetrics.repairDelayedByReplica::getCount,
+                            5,
+                            TimeUnit.MINUTES);
+            Util.spinAssert("AutoRepair has not observed any schedule contention in INCREMENTAL repair",
                             greaterThan(0L),
-                            AutoRepair.instance.repairStates.get(AutoRepairConfig.RepairType.INCREMENTAL)::getLastRepairTime,
+                            incrementalMetrics.repairDelayedBySchedule::getCount,
                             5,
                             TimeUnit.MINUTES);
+
+            AutoRepairMetrics fullMetrics = AutoRepairMetricsManager.getMetrics(AutoRepairConfig.RepairType.FULL);
+            Util.spinAssert("AutoRepair has not yet completed one FULL repair cycle",
```

Comment:
Checking full repair after incremental repair since I expect IR to be the long poll since we wait for it to produce some contention.